### ISSUE TITLE
How to handle file name with “|” symbol in windows system

### ISSUE THREAD

**Issue by GaoJianAllen** [cookiecutter/cookiecutter#1632]
* Cookiecutter version: latest
* Template project url:
* Python version:
* Operating System: Windows

### Description:

I have a template contains file name with "|" like **{{cookiecutter.service_name|lower}}.json** .
Obviously, it can't work in windows system, because windows can't allow filename contains "|".
Does cookiecutter have other replacement or workaround for this?
It works well in macOS and Linux.

**Comment by GaoJianAllen** [comment]
Any update?

**Comment by luzfcb** [comment]
@GaoJianAllen 

Try to use private variables:

https://cookiecutter.readthedocs.io/en/latest/advanced/private_variables.html

cookiecutter.json
```json
{
    "service_name": "My Service",
    "__service_name": "{{ cookiecutter.service_name|lower|replace(' ', '_') }}"
}
```

**Comment by luzfcb** [comment]
@GaoJianAllen 

Try to use private variables.

https://cookiecutter.readthedocs.io/en/latest/advanced/private_variables.html

cookiecutter.json
```json
{
    "service_name": "My Service",
    "__service_name": "{{ cookiecutter.service_name|lower|replace(' ', '_') }}"
}
```

and define the filename like:

```bash
{{cookiecutter.__service_name}}.json
```

Note: **Private variable is available only in cookiecutter >=2.0 (not yet released).** 
You can install the latest version from the master branch using

```bash
python3 -m pip install cookiecutter@https://github.com/cookiecutter/cookiecutter/archive/master.zip
```

**Comment by GaoJianAllen** [comment]
@luzfcb Thanks. It works now. Looking forward to the new release 2.0
